File: src/main/java/com/cfgglobal/test/service/rule/access/IbAccessRule.java
package com.cfgglobal.test.service.rule.access;


import com.cfgglobal.test.base.Filter;
import com.cfgglobal.test.domain.Permission;
import com.cfgglobal.test.domain.User;
import org.springframework.stereotype.Component;

import static com.cfgglobal.test.base.Filter.OPERATOR_EQ;
import static com.cfgglobal.test.base.Filter.OPERATOR_IN;

@Component
public class IbAccessRule extends AbstractAccessRule {
    @Override
    public Filter exec(Permission permission) {
        User user = securityFilter.currentUser();
        Filter filter = new Filter();
        if (permission.getEntity().equals("user")) {
            filter.addCondition("introducedBy.id", user.getId(), OPERATOR_EQ);
        } else if (permission.getEntity().equals("price-category")) {
            filter.addCondition("id", new Object[]{5, 6, 7, 9}, OPERATOR_IN);
        } else {
            filter.addCondition("user.introducedBy.id", user.getId(), OPERATOR_EQ);
        }

        return filter;
    }

    @Override
    public String getRuleName() {
        return "ib";
    }
}

File: src/main/kotlin/com/cfgglobal/test/service/rule/access/IbAccessRule.kt
package com.cfgglobal.test.service.rule.access


import com.cfgglobal.test.domain.Permission
import com.cfgglobal.test.vo.Filter
import com.cfgglobal.test.vo.Filter.Companion.OPERATOR_EQ
import com.cfgglobal.test.vo.Filter.Companion.OPERATOR_IN

import org.springframework.stereotype.Component

@Component
class IbAccessRule : AbstractAccessRule() {

    override val ruleName: String
        get() = "ib"

    override fun exec(permission: Permission): Filter {
        val user = securityFilter!!.currentUser()
        val filter = Filter()
        when {
            permission.entity == "user" -> filter.addCondition("introducedBy.id", user.id, OPERATOR_EQ)
            permission.entity == "price-category" -> filter.addCondition("id", arrayOf(5, 6, 7, 9), OPERATOR_IN)
            else -> filter.addCondition("user.introducedBy.id", user.id, OPERATOR_EQ)
        }

        return filter
    }
}
